scripts/halium

```diff
@@ -430,36 +430,58 @@ mountroot() {
 	# Make sure the device has been created by udev before we try to mount
 	udevadm settle
 
-	# find the right partition
-	for partname in $partlist; do
-		part=$(find /dev -name $partname | tail -1)
-		[ -z "$part" ] && continue
-		path=$(readlink -f $part)
-		[ -n "$path" ] && break
-	done
+	# Check for LVM volume group in kernel cmdline and activate LVM
+	vg=""
+	if grep -q lvm_vg= /proc/cmdline; then
+		for x in $(cat /proc/cmdline); do
+			case ${x} in
+			lvm_vg=*)
+				vg=${x#*=}
+				;;
+			esac
+		done
+	fi
```

Comment:
- I think the `grep ...` is not really needed, as if there isn't this parameter then the loop below won't find it anyway.
- Maybe simply `lvm_vg` is too generic. Maybe `halium_lvm_vg`, but I don't know...